plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,808 @@
+    lldp_receive_state = module.params.get("lldp_receive_state")
+    stp_bpdu_filter = module.params.get("stp_bpdu_filter")
+    stp_bpdu_guard = module.params.get("stp_bpdu_guard")
+    llfc_transmit_state = module.params.get("llfc_transmit_state")
+    llfc_receive_state = module.params.get("llfc_receive_state")
+    mcp_admin_state = module.params.get("mcp_admin_state")
+    mcp_strict_mode = module.params.get("mcp_strict_mode")
+    mcp_initial_delay_time = module.params.get("mcp_initial_delay_time")
+    mcp_transmission_frequency_sec = module.params.get("mcp_transmission_frequency_sec")
+    mcp_transmission_frequency_msec = module.params.get("mcp_transmission_frequency_msec")
+    mcp_grace_period_sec = module.params.get("mcp_grace_period_sec")
+    mcp_grace_period_msec = module.params.get("mcp_grace_period_msec")
+    pfc_admin_state = module.params.get("pfc_admin_state")
+    access_mac_sec_policy = module.params.get("access_mac_sec_policy")
+    state = module.params.get("state")
+
+    ops = []
+    match = None
+
+    mso_template = MSOTemplate(mso, "fabric_policy", template)
+    mso_template.validate_template("fabricPolicy")
+
+    path = "/fabricPolicyTemplate/template/interfacePolicyGroups"
+    object_description = "Interface Policy Groups"
+
+    template_info = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {})
+
+    existing_interface_policies = mso_template.template.get("fabricPolicyTemplate", {}).get("template", {}).get("interfacePolicyGroups", [])
+    if interface_policy_group or interface_policy_group_uuid:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_interface_policies,
+            [KVPair("uuid", interface_policy_group_uuid) if interface_policy_group_uuid else KVPair("name", interface_policy_group)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_interface_policies
+
+    if state == "present":
+
+        if match:
+            if interface_policy_group and match.details.get("name") != interface_policy_group:
+                ops.append(dict(op="replace", path="{0}/{1}/name".format(path, match.index), value=interface_policy_group))
+                match.details["name"] = interface_policy_group
+
+            if description is not None and match.details.get("description") != description:
+                ops.append(dict(op="replace", path="{0}/{1}/description".format(path, match.index), value=description))
+                match.details["description"] = description
+
+            if interface_type and match.details.get("type") != interface_type:
+                mso.fail_json(msg="Interface type cannot be changed")
+
+            if domains:
+                # the existing domain list is updated by the new domain list
+                domain_list = []
```

Comment:
can we rename this to uuids, to make it easier to follow we are setting the uuids
```suggestion
                domain_uuids = []
```